Autopilot/boardfiles/Src/freertos.c

```diff
@@ -148,28 +158,69 @@ void MX_FREERTOS_Init(void) {
   osThreadDef(Attitude, Attitude_Run, osPriorityNormal, 0, 128);
   AttitudeHandle = osThreadCreate(osThread(Attitude), NULL);
 
+  /* definition and creation of pathManager */
+  osThreadDef(pathManager, pathManagerExecute, osPriorityBelowNormal, 0, 128);
+  pathManagerHandle = osThreadCreate(osThread(pathManager), NULL);
+
+  /* definition and creation of telemetryRun */
+  osThreadDef(telemetryRun, StartTelemetryRun, osPriorityBelowNormal, 0, 128);
+  telemetryRunHandle = osThreadCreate(osThread(telemetryRun), NULL);
+
   /* USER CODE BEGIN RTOS_THREADS */
   /* add threads, ... */
   /* USER CODE END RTOS_THREADS */
 
 }
 
-/* USER CODE BEGIN Header_StartDefaultTask */
+/* USER CODE BEGIN Header_attitudeManagerExecute */
 /**
-  * @brief  Function implementing the defaultTask thread.
+  * @brief  Function implementing the attitudeManager thread.
   * @param  argument: Not used
   * @retval None
   */
-/* USER CODE END Header_StartDefaultTask */
-void StartDefaultTask(void const * argument)
+/* USER CODE END Header_attitudeManagerExecute */
+void attitudeManagerExecute(void const * argument)
+{
+  /* USER CODE BEGIN attitudeManagerExecute */
+  /* Infinite loop */
+  TickType_t xLastWakeTime = xTaskGetTickCount();
+  vTaskDelayUntil(&xLastWakeTime, PERIOD_ATTITUDEMANAGER_MS);
+  AttitudeManagerInterfaceExecute();
```

Comment:
About this, the whole state machine needs to execute a full cycle everytime this is called. Currently, this will only execute 1 state. 

You'll need to put this in a for loop or something and keep calling execute until the status reads done, or god forbid, failed. (Look at the attitude manager header for info about the status).

The same will be true for path and telemetry.